metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py

```diff
@@ -126,24 +137,80 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         config = DeltaLakeSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    inner_data_type = ""
+    counter = 0
+
+    def _get_inner_field_details(self, raw_field):
```

Comment:
Looks like this parses json form returned by deltalake library to hive-like form. Can we change this method to be more declarative and self-explanatory, for example:
```
if raw_field["type"]=="array":
    # parse elementType .
elif raw_field["type"]=="struct":
    # parse fields
```

This could be effectively similar to [this](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/trino.py#L239) method in structure.